src/main/java/net/querz/nbt/ListTag.java

```diff
@@ -8,16 +8,22 @@
 import java.util.List;
 import java.util.Objects;
 
+import static net.querz.nbt.Tag.Type.*;
+
 public non-sealed class ListTag extends CollectionTag<Tag> {
 
 	private final List<Tag> value;
-	private byte type;
+	private Type type;
 
 	public ListTag() {
-		this(new ArrayList<>(), END);
+		this(new ArrayList<>(), null);
 	}
 
-	public ListTag(List<Tag> list, byte type) {
+	public ListTag(List<Tag> list, Type type) {
+		if (type == END) {
+			throw new IllegalArgumentException("ListTag can not be of type END");
```

Comment:
Why not? empty ListTags are of type END, that's what is being written in `ListTag#write`.

Reply:
I see, makes a lot more sense this way.